Loop/Managers/DeviceDataManager.swift

```diff
@@ -664,24 +664,25 @@ extension DeviceDataManager {
             return
         }
 
-        self.loopManager.addRequestedBolus(DoseEntry(type: .bolus, startDate: Date(), value: units, unit: .units), completion: nil)
-        pumpManager.enactBolus(units: units, at: startDate) { (result) in
-            switch result {
-            case .failure(let error):
-                self.log.error("%{public}@", String(describing: error))
-                switch error {
-                case .uncertainDelivery:
-                    // Do not generate notification on uncertain delivery error
-                    break
-                default:
-                    NotificationManager.sendBolusFailureNotification(for: error, units: units, at: startDate)
-                }
-                self.loopManager.bolusRequestFailed(error) {
-                    completion(error)
-                }
-            case .success(let dose):
-                self.loopManager.bolusConfirmed(dose) {
-                    completion(nil)
+        self.loopManager.addRequestedBolus(DoseEntry(type: .bolus, startDate: Date(), value: units, unit: .units)) {
```

Comment:
TBH I'm not 100% sure this is necessary for this fix, but it looked wrong to me to not wait for `addRequestedBolus` to call its `completion`.  Looking 👀 at @ps2 to check this work...